applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -94,23 +96,23 @@ public static Builder builder() {
      * If {@link ConfigurationSetting#etag()} is specified, the configuration value is updated if the current setting's
      * etag matches. If the etag's value is equal to {@link ConfigurationClient#ETAG_ANY}, the setting will always be
      * updated.
-     * </p>
      *
      * <p>
-     * The label value for the ConfigurationSetting is optional. If not specified, the
+     * The label value for the ConfigurationSetting is optional. If not specified,
      * {@link ConfigurationSetting#NULL_LABEL} is used.
-     * </p>
      *
-     * @param configurationSetting The configuration setting to create or update.
+     * @param setting The configuration setting to create or updateSetting.
      * @return ConfigurationSetting that was created or updated.
+     * @throws NullPointerException                     If {@code setting} is {@code null}.
+     * @throws IllegalArgumentException                 If {@link ConfigurationSetting#key()} is {@code null} or {@code ""}.
      * @throws com.azure.common.http.rest.RestException If the {@link ConfigurationSetting#etag()} was specified, is not
-     *                                               {@link ConfigurationClient#ETAG_ANY}, and the current configuration
-     *                                               value's etag does not match.
+     *                                                  {@link ConfigurationClient#ETAG_ANY}, and the current configuration
+     *                                                  value's etag does not match.
```

Comment:
Don't worry about this kind of alignment either - just wrap back to start of line - maybe indent 4 spaces or something, but don't try to do this, it's just grounds for a wall of text (which is all squished into a narrow column).

Reply:
Sounds good. I updated my IntelliJ format settings to stop doing it.